src/App.js

```diff
@@ -10,10 +10,138 @@ function App() {
     beginDate: '',
     endDate: ''
   });
-  const [glocationFilter, setGlocationFilter] = useState('');
-  const [newsDeskFilter, setNewsDeskFilter] = useState([]);
-  const [materialTypeFilter, setMaterialTypeFilter] = useState([]);
+  const [glocation, setGlocation] = useState('');
+  const [newsDesks, setNewsDesks] = useState([]);
+  const [materialTypes, setMaterialTypes] = useState([]);
   const [sortOrder, setSortOrder] = useState('relevance');
+  const [articles, setArticles] = useState(null);
+  const [totalHits, setTotalHits] = useState(0);
+  const [page, setPage] = useState(0);
+  const [isFetching, setIsFetching] = useState(false);
+  
+  const isMounted = useRef(false);
+
+  // If page changes, submit search
+  // Do nothing on initial render
+  useEffect(() => {
+    if (isMounted.current) {
+      fetchArticles();
+    }
+  }, [page]);
+  
+  // If `sortOrder` changes, either set page back to 0 or submit search
+  // Do nothing on initial render
+  useEffect(() => {
+    if (isMounted.current) {
+      if (page !== 0) {
+        setPage(0)
+      } else {
+        fetchArticles();
+      }
+    }
+  }, [sortOrder]);
+
+  // On initial render, set `isMounted` equal to true
+  useEffect(() => {
+    if (!isMounted.current) {
+      isMounted.current = true;
+    }
+  });
+
+  // When SEARCH button is clicked, either reset `sortOrder` or `page`, or submit a search
+  const submitNewSearch = () => {
+    if (sortOrder !== 'relevance') {
+      setSortOrder('relevance');
+    } else if (page !== 0) {
+      setPage(0);
+    } else {
+      fetchArticles();
+    }
+  }
+
+  const fetchArticles = async () => {
+    const baseURL = 'https://api.nytimes.com/svc/search/v2/articlesearch.json';
+    const key = 'brtQ9fXA0I1ATPctklZe6RcanXZRklYl';
+    let fullURL = `${baseURL}?api-key=${key}&page=${page}&sort=${sortOrder}`;
+
+    fullURL += searchQuery.query ? `&q=${searchQuery.query}` : '';
+    fullURL += searchQuery.beginDate ? `&begin_date=${searchQuery.beginDate}` : '';
+    fullURL += searchQuery.endDate ? `&end_date=${searchQuery.endDate}` : '';
+
+    let activeFilters = getActiveFiltersForFetchURL();
+    fullURL += activeFilters.length > 0 ? `&fq=${activeFilters.join(' AND ')}` : '';
+
+    setIsFetching(true);
+    const response = await fetch(fullURL);
+    const searchResults = await response.json();
+    setArticles(searchResults.response.docs);
+    setTotalHits(searchResults.response.meta.hits);
+    setIsFetching(false);
+  }
+
+  // Encode all active filter fields and values and return them 
+  // in an array for insertion into the API fetch URL
+  const getActiveFiltersForFetchURL = () => {
```

Comment:
mmm, this functionality seems like something that the component itself should maintain. however, I think React does not really encourage this sort of encapsulation. so nevermind this comment :)